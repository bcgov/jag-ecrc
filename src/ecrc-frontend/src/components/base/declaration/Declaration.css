.declareTitle {
  width: 100%;
  height: 36px;
  background: inherit;
  background-color: rgba(56, 89, 138, 1);
  box-sizing: border-box;
  border-width: 3px;
  border-style: solid;
  font-weight: 400;
  color: white;
  padding-left: 20px;
  padding-top: 2px;
  vertical-align: middle;
}

.declaration-cb {
  font-size: 14px;
  color: #333333;
  text-align: left;
  line-height: 20px;
}

.declareSection {
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 20px;
}
