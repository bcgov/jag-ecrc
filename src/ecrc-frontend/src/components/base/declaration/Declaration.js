import React from "react";
import PropTypes from "prop-types";
import { TextInput } from "../textInput/TextInput";
import "./Declaration.css";

export default function Declaration({ onApplicantNameChange }) {
  const textInput = {
    label: "Applicant Name",
    id: "textInputId",
    textInputStyle: "textinput_editable_white"
  };

  return (
    <div>
      <div className="declareTitle">DECLARATION AND CONSENT</div>
      <section className="declareSection">
        <input type="checkbox" />
        <span className="declaration-cb">
          I, the undersigned, do hereby consent to the collection and disclosure
          by the Royal Canadian Mounted Police (RCMP) and other law enforcement
          agencies, as well as other duly authorized agencies of the government,
          of any and all information related to the security screening checks in
          support of this application for up to two years. This may include some
          or all of: (a) criminal record check or fingerprint-based criminal
          record verification by searching the Canadian Police Information
          Centre database; (b) a police information check; (c) a check of
          intelligence databases maintained by law enforcement agencies; (d) a
          check of records in the justice information system of the Ministry of
          Attorney General and; (e) a check of records in the corrections
          information system of the Ministry of Public Safety and Solicitor
          General.
        </span>
      </section>
      <section className="declareSection">
        <input type="checkbox" />
        <span className="declaration-cb">
          I certify that, to the best of my knowledge, the information I have
          provided on my application and will provide as necessary is complete,
          honest and accurate. I understand that a false statement or omission
          of facts herein may lead to a denial of a cannabis workers
          registration. I am also aware that later discovery of an omission or
          misrepresentation may be grounds for any finding of suitability to be
          suspended or revoked.
        </span>
      </section>

      <section className="declareSection" style={{ width: "475px" }}>
        <TextInput textInput={textInput} onChange={onApplicantNameChange} />
      </section>
    </div>
  );
}

Declaration.propTypes = {
  onApplicantNameChange: PropTypes.func.isRequired
};
