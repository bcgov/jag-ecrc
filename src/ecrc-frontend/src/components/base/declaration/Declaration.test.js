/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { create } from "react-test-renderer";

import Declaration from "./Declaration";

describe("Declaration Component", () => {
  test("Matches the default snapshot", () => {
    const declaration = create(<Declaration shareConsent={false} />);
    expect(declaration.toJSON()).toMatchSnapshot();
  });
  test("Matches the sharing snapshot", () => {
    const declaration = create(<Declaration shareConsent />);
    expect(declaration.toJSON()).toMatchSnapshot();
  });
});
