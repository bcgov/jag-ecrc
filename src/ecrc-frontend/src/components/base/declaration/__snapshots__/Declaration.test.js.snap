// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Declaration Component Matches the snapshot 1`] = `
<div>
  <div
    className="declareTitle"
  >
    DECLARATION AND CONSENT
  </div>
  <section
    className="declareSection"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="declaration-cb"
    >
      I, the undersigned, do hereby consent to the collection and disclosure by the Royal Canadian Mounted Police (RCMP) and other law enforcement agencies, as well as other duly authorized agencies of the government, of any and all information related to the security screening checks in support of this application for up to two years. This may include some or all of: (a) criminal record check or fingerprint-based criminal record verification by searching the Canadian Police Information Centre database; (b) a police information check; (c) a check of intelligence databases maintained by law enforcement agencies; (d) a check of records in the justice information system of the Ministry of Attorney General and; (e) a check of records in the corrections information system of the Ministry of Public Safety and Solicitor General.
    </span>
  </section>
  <section
    className="declareSection"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="declaration-cb"
    >
      I certify that, to the best of my knowledge, the information I have provided on my application and will provide as necessary is complete, honest and accurate. I understand that a false statement or omission of facts herein may lead to a denial of a cannabis workers registration. I am also aware that later discovery of an omission or misrepresentation may be grounds for any finding of suitability to be suspended or revoked.
    </span>
  </section>
  <section
    className="declareSection"
  >
    <input
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="declaration-cb"
    >
      The third checkbox for my application and will provide as necessary is complete.
    </span>
  </section>
  <section
    className="declareSection"
    style={
      Object {
        "maxWidth": "475px",
      }
    }
  >
    <div>
      <div
        className="label"
      >
        <label
          className="textinput_label"
          htmlFor=""
        >
          
        </label>
        
        Â 
        <span
          className="note"
        />
      </div>
      <input
        className="textinput_editable_white"
        id=""
        onChange={[Function]}
        type="text"
      />
    </div>
  </section>
</div>
`;
